File: RobWork/example/cpp/ex-quaternion.cpp
#include <rw/math/Quaternion.hpp>
#include <rw/math/Rotation3D.hpp>

using namespace rw::math;

int main (int argc, char** argv)
{
    const Quaternion<> quat = Quaternion<> (sqrt (2) / 2, sqrt (2) / 2, 0, 0);
    std::cout << "Quaternion: " << quat << std::endl;
    const Rotation3D<> rotationFromQuat = quat.toRotation3D ();
    std::cout << "Rotation from Quaternion: " << rotationFromQuat << std::endl;

    const Rotation3D<> rot = Rotation3D<> (-1, 0, 0, 0, 0, 1, 0, 1, 0);
    std::cout << "Rotation: " << rot << std::endl;
    const Quaternion<> quatFromRotation = Quaternion<> (rot);
    std::cout << "Quaternion from Rotation: " << quatFromRotation << std::endl;

    return 0;
}

File: RobWork/example/java/src/ExQuaternion.java
import org.robwork.LoaderRW;
import org.robwork.sdurw.*;
import java.lang.Math;

public class ExQuaternion {
    public static void main(String[] args) throws Exception {
        LoaderRW.load("sdurw");

        Quaterniond quat = new Quaterniond(Math.sqrt(2)/2,Math.sqrt(2)/2,0,0);
        System.out.println("Quaternion: " + quat);
        Rotation3Dd rotationFromQuat = quat.toRotation3D();
        System.out.println("Rotation from Quaternion: " + rotationFromQuat);

        Rotation3Dd rot = new Rotation3Dd(-1,0,0,0,0,1,0,1,0);
        System.out.println("Rotation: " + rot);
        Quaterniond quatFromRotation = new Quaterniond(rot);
        System.out.println("Quaternion from Rotation: " + quatFromRotation);
    }
}
